webapp/platform/components/src/compass-ui/components/modal/generic_modal.tsx

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo} from 'react';
+import {styled} from '@mui/material/styles';
+import Button from '@mui/material/Button';
+import MUIPaper, {PaperProps} from '@mui/material/Paper';
+import DialogActions from '@mui/material/DialogActions';
+import Slide from '@mui/material/Slide';
+import {TransitionProps} from '@mui/material/transitions';
+import BaseModal from "../base_modal/base_modal";
+
+const Transition = React.forwardRef((
+    {children, ...props}: TransitionProps & {
+        children: React.ReactElement<any, any>;
+    },
+    ref: React.Ref<unknown>,
+) => {
+    return (
+        <Slide
+            direction='up'
+            ref={ref}
+            {...props}
+        >
+            {children}
+        </Slide>
+    );
+});
+
+const Paper = styled(MUIPaper)<PaperProps>(() => ({
+    border: '1px solid rgba(var(--center-channel-color-rgb), 0.16)',
+    borderRadius: 12,
+    backgroundColor: 'var(--center-channel-bg)',
+    boxShadow: '0 20px 32px 0 rgba(0, 0, 0, 0.12)',
+    minWidth: 600,
+}));
+
+type ModalProps = {
+    children: React.ReactNode | React.ReactNodeArray;
+    isOpen: boolean;
+    dialogClassName?: string;
+    dialogId?: string;
+    onClose?: () => void;
+    onConfirm?: () => void;
+    onCancel?: () => void;
+}
+
+const GenericModal = ({children, isOpen, onClose, onConfirm, onCancel, dialogClassName = '', dialogId = ''}: ModalProps) => {
+    const hasActions = Boolean(onConfirm || onCancel);
+
+    const handleConfirmAction = () => {
+        onConfirm?.();
+        onClose?.();
+    };
+
+    const handleCancelAction = () => {
+        onCancel?.();
+        onClose?.();
+    };
+
+    return (
+            <BaseModal
+                isOpen={isOpen}
+                onClose={onClose}
+                aria-describedby='alert-dialog-slide-description'
+                dialogClassName={dialogClassName}
+                dialogId={dialogId}
+            >
+                {children}
+                {hasActions && (
+                    <DialogActions>
+                        {onCancel && <Button onClick={handleCancelAction}>{'Cancel'}</Button>}
```

Comment:
assuming these will be translated?

Reply:
@hmhealey any suggestions for this?

Reply:
You're able to use react-intl here, but i18n-extract isn't currently able to find the translations on its own, so you'd need to add a `t` function somewhere in `channels` with this string. We've already done that for the current `GenericModal` [here](https://github.com/mattermost/mattermost-server/blob/master/webapp/channels/src/components/generic_modal.tsx#L10-L16), so you can probably just reuse those same IDs here